test/Change.lean

```diff
@@ -0,0 +1,20 @@
+import Mathlib.Tactic.Change
+import Std.Tactic.GuardMsgs
+
+set_option pp.unicode.fun true
+
+/--
+info: Try this: change 0 = 1
+---
+info: Try this: change (fun x ↦ x) 0 = 1
+---
+info: Try this: change (fun x ↦ x) 0 = 1
+---
+error: The given term is not DefEq to the goal
+-/
+#guard_msgs in
+example : (fun x : Nat => x) 0 = 1 := by
+  change? 0 = _  -- change 0 = 1
+  change?        -- change (fun x ↦ x) 0 = 1
+  change? _      -- change (fun x ↦ x) 0 = 1
+  change? 1 = _  -- The given term is not DefEq to the goal
```

Comment:
Note: we don't use `_` because it becomes `?m.96`, and that's quite fragile to put in a `#guard_msgs`
```suggestion
error: The term
  1 = 0
is not defeq to the goal:
  (fun x ↦ x) 0 = 1
-/
#guard_msgs in
example : (fun x : Nat => x) 0 = 1 := by
  change? 0 = _  -- change 0 = 1
  change?        -- change (fun x ↦ x) 0 = 1
  change? _      -- change (fun x ↦ x) 0 = 1
  change? 1 = 0
    -- The term
    --   1 = 0
    -- is not defeq to the goal:
    --   (fun x ↦ x) 0 = 1
```